prober/notification.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package prober
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"expvar"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+var (
+	alertGenerated   = expvar.NewInt("alert_generated")
+	alertFailed      = expvar.NewInt("alert_failed")
+	warningGenerated = expvar.NewInt("warning_generated")
+	warningFailed    = expvar.NewInt("warning_failed")
+)
+
+// sendAlert sends an alert to the incident response system, to
+// page a human responder immediately.
+// summary should be short and state the nature of the emergency.
+// details can be longer, up to 29 KBytes.
+func sendAlert(summary, details string) error {
+	type squadcastAlert struct {
+		Message     string            `json:"message"`
+		Description string            `json:"description"`
+		Tags        map[string]string `json:"tags,omitempty"`
+		Status      string            `json:"status"`
+		EventId     string            `json:"event_id"`
+	}
+
+	tags := make(map[string]string)
+	tags["severity"] = "critical"
+	sqa := squadcastAlert{
+		Message:     summary,
+		Description: details,
+		Tags:        tags,
+		Status:      "trigger",
+		EventId:     uuid.New().String(),
+	}
+
+	sqBody, err := json.Marshal(sqa)
+	if err != nil {
+		alertFailed.Add(1)
+		return err
+	}
+
+	webhookUrl := os.Getenv("SQUADCAST_WEBHOOK")
+	if webhookUrl == "" {
+		warningFailed.Add(1)
+		return errors.New("No SQUADCAST_WEBHOOK configured")
+	}
+
+	req, err := http.NewRequest(http.MethodPost, webhookUrl, bytes.NewBuffer(sqBody))
+	if err != nil {
+		alertFailed.Add(1)
+		return err
+	}
+
+	req.Header.Add("Content-Type", "application/json")
```

Comment:
```suggestion
	req.Header.Set("Content-Type", "application/json")
```

There shouldn't be one yet in this context anyway, though.